### PR TITLE
REST - Add overwrite logic for parameters

### PR DESCRIPTION
**Pull request by hellosnow**
@ansyral @chenkennt @vicancy @vwxyzh @superyyrrzz @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Common/EntityMergers/JObjectMerger.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public class JObjectMerger : MergerDecorator
+    {
+        public JObjectMerger(IMerger inner)
+            : base(inner)
+        {
+        }
+
+        public override void Merge(ref object source, object overrides, Type type, IMergeContext context)
+        {
+            var stopMerge = false;
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/EntityMergers/JObjectMerger.cs]: add `if (type == typeof(object))`?

File: src/Microsoft.DocAsCode.Build.Common/ApplyOverwriteDocument.cs [hunk 2]
```diff
@@ -14,10 +14,10 @@ namespace Microsoft.DocAsCode.Build.Common
 
     public abstract class ApplyOverwriteDocument : BaseDocumentBuildStep
     {
-        private readonly MergerFacade Merger = new MergerFacade(
-                new DictionaryMerger(
-                    new KeyedListMerger(
-                        new ReflectionEntityMerger())));
+        protected virtual MergerFacade Merger { get; set; } = new MergerFacade(
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Common/ApplyOverwriteDocument.cs]: `get` only?

Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Common/ApplyOverwriteDocument.cs]: >protected virtual [](start = 8, length = 17)

how about use MergerFacade factory classes?

Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/ApplyOverwriteDocument.cs]: Add todo and consider to add later

---
In reply to: [91022771](https://github.com/dotnet/docfx/pull/965#discussion_r91022771) [](ancestors = 91022771)

**Pull request merged**